### PR TITLE
Improve WC & Twenty twenty compatibility

### LINKED ISSUE #24648: WooCommerce support for Twenty Twenty theme
WordPress 5.3 will include a new core theme that will be released on November 12. Will WooCommerce support this theme?

### PR DESCRIPTION
**Pull request by peterfabian**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #24648  .

Todos to fix:
* [x] Wide layout for single product
* [x] Middle layout for single product
* [x] Narrow layout for single product

* [x] Wide shop layout 
* [x] Mid shop layout 
* [x] Narrow shop layout 

* [x] Wide cart layout 
* [x] Mid cart layout 
* [x] Narrow cart layout 

* [x] Wide checkout layout 
* [x] Mid checkout layout 
* [x] Narrow checkout layout 

* [x] Wide My account layout 
* [x] Mid My account layout 
* [x] Narrow My account layout 

* [x] Wide Homepage layout 
* [x] Mid Homepage layout 
* [x] Narrow Homepage layout 

* [x] Select2

* [x] Log in form on My Account page
* [x] Log in form on Checkout page

Also, [smooth scroll](https://core.trac.wordpress.org/ticket/48551) is annoying, but I'm going to believe it's gonna be sorted out in the Twenty twenty theme itself.

Screenshots:

Category narrow view:
Category_view_narrow
Category view wide:
Category_view_wide

My Account Orders narrow:
MyAccount_Orders_narrow

My Account Orders wide:
MyAccount_Orders

My Account logged out:
MyAccount_logged_out

Thank you narrow view:
ThankYou_narrow

Thank you wide view:
ThankYou_wide

Checkout narrow view:
Checkout_narrow

Checkout wide view:
Checkout_wide

Cart narrow view:
Cart_narrow

Cart wide view:
Cart_wide

Single product with desc and reviews:
Single_product_desc_review

Single product with desc and reviews:
Single_product_desc_reviews

Single product with no desc and no reviews narrow view:
Single_product_no_desc_no_rating_narrow

Single product with no desc and no reviews wide view:
Single_product_no_desc_no_rating_wide

Shop wide view:
Shop_wide

Homepage narrow view:
Homepaeg_narrow

Homepage  wide view:
Homepage_wide

### How to test the changes in this Pull Request:

Apply Twenty twenty theme and this PR, test that everything looks non-broken, among other things:

- Homepage made out of blocks (as from Storefront) in wide and mobile layout (mobile layout preserves number of columns with some blocks like Top rated products, but it's the same for Storefront and 2019, so I let it be), try adding to cart
- shop page in wide and mobile layout, try adding to cart
- single product page, try adding to cart, try variable product, with or without rating, observer the annoying scroll to behaviour when switching tabs
- cart page wide and narrow, updating cart, empty cart
- checkout page wide and narrow, submit with data missing, submit correctly
- Thank you for order page
- My Account subpages
- Select2 on checkout page and cart pages
- logged in and out]
- try creating an account via My Account, logging in via Checkout

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Improved WooCommerce and Twenty twenty styling.

### CODE REVIEW

**Review by vedanshujain — APPROVED** [review]
Looks good. Left some minor comment but not blocker

File: assets/css/twenty-twenty.scss [hunk 1]
```diff
@@ -0,0 +1,2596 @@
...
+.woocommerce-product-gallery--columns-4 {
+
+	ol {
+		margin-left: 0;
+		margin-bottom: 0;
+	}
+
+	.flex-control-thumbs li {
+		width: 14.2857142857%;
+		margin: 0 14.2857142857% 1.6em 0;
```
[the remark this answers is not in the document]
Comment (peterfabian) [reply assets/css/twenty-twenty.scss]: It's inspired by Storefront, so I guess there might be a reason, I'm not totally sure it's required, though.

File: includes/theme-support/class-wc-twenty-twenty.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	 */
+	public static function output_content_wrapper_end() {
+		echo '</main>';
+		echo '</section>';
+	}
+
+	/**
+	 * Set background color to white if it's default, otherwise don't touch it.
+	 */
+	public static function set_white_background() {
```
[the remark this answers is not in the document]
Comment (peterfabian) [reply includes/theme-support/class-wc-twenty-twenty.php]: I wanted to retain the functionality of people choosing their bg color, but switch the default to white. If I only overrode this in CSS, the both block editor and tinymce still used the predefined background color, and I don't think that is correct.

**Comment by kloon** [comment]
The last product on a new row in the product list on the shop page should be align left. It seem this is trying to align center, but at present it is also not really aligning center correctly.

**Comment by kloon** [comment]
On the single product page, there's an underline on the  magnifying glass

**Comment by kloon** [comment]
Checkout page is set to full width, however it is smushing eveything together and boxes the page in

**Comment by peterfabian** [comment]
Updated the related products. 
Can't get the same errors with zoom icon and checkout, unfortunately.

**Comment by kloon** [comment]
I tested this with Chrome Dev 79 which was giving the checkout and hourglass errors, when I test with Safari and Chrome Stable it works fine so we can ignore the issues for now but will definitely need to create a follow up to sort it out since Chrome 79 goes stable in a couple of weeks.

**Review by kloon — APPROVED** [review]
The latest changes with the related product fix does the job. We will need to create a follow up issue to address the Chrome Beta styling issues for a future release though.

**Comment by peterfabian** [comment]
Thanks! I've tested with Chrome 79 beta and could not reproduce, so we might need to dig a bit more into what's causing it:
Screenshot 2019-11-27 at 16 33 04

File: includes/theme-support/class-wc-twenty-twenty.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (carlomanf) [reply includes/theme-support/class-wc-twenty-twenty.php:70]: This needs to be changed. It literally singles out a specific colour, bans the user from selecting it and forcefully replaces it with white. Better not let this code get anywhere near the skin tone emoji.

**Pull request merged**